ptx_parser/src/ast.rs

```diff
@@ -1138,6 +1138,40 @@ impl<Ident> ParsedOperand<Ident> {
     }
 }
 
+impl<Ident> std::fmt::Display for ParsedOperand<Ident>
+where
+    Ident: std::fmt::Display,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ParsedOperand::Reg(id) => write!(f, "{}", id)?,
+            ParsedOperand::RegOffset(id, o) => write!(f, "[{}+{}]", id, o)?,
```

Comment:
`[` `]` feel unnecessary here, AFAIK grammar does not prevent RegOffset from showing in non-bracketized positions. Though it might might make it more readable in real world?
I'm leaving it up to you